src/world/providers/anvil/Region.ts

```diff
@@ -26,32 +29,52 @@ export default class Region {
 
         // TODO: timestamps
 
-        locations.forEach((location) => {
-            // FIXME: This should be replaced with stream.setOffset in the future
-            const stream = new BinaryStream(buffer, location[0] * 4096);
-            for (let i = 0; i < location[1]; i++) {
-                const sector = new BinaryStream(stream.read(4096));
+        for (const location of locations) {
+            (stream as any).offset = location[0] * 4096;
+            for (let j = 0; j < location[1]; j++) {
+                const size = stream.readInt();
+                const compression = stream.readByte();
+                const sector = stream.read(4091);
 
-                const size = sector.readInt();
-                const compression = sector.readByte();
                 let data;
                 switch (compression) {
                     case AnvilChunkCompressionType.Zlib:
-                        data = Zlib.unzipSync(sector.read(size - 1));
+                        data = Zlib.unzipSync(sector.slice(0, size - 1));
                         break;
                     case AnvilChunkCompressionType.Uncompressed:
-                        data = sector.read(size - 1);
+                        data = sector.slice(0, size - 1);
                         break;
                     default:
                         throw new Error(`Invalid compression type`);
                 }
 
                 const nbtStream = new BinaryStream(data);
-                const nbt = NBTTagCompound.readFromStream(
+                const regionNbt = NBTTagCompound.readFromStream(
                     nbtStream,
                     ByteOrder.BIG_ENDIAN
                 );
+
+                if (regionNbt.has('Level')) {
```

Comment:
Makes more sense to do a:
```typescript
if (!regionNbt.has('Level'))
    throw new Error('region is missing Level tag');
```
instead of incasing the logic in a new scope.